projecttacoma/measure-repository
# Creatable Artifacts

# Summary
Adds new authoring page and ability to create simple Library and Measure draft resources and store them in a trpc+mongo backend.

## New behavior
New Authoring page!!
This page gives you the ability to create new draft Libraries and Measures, as well as enter some custom simple information, like url and identifier

Our trpc backend server is now connected to a mongo database.

## Code changes
 - Created Authoring page and added Authoring button to home screen nav bar
 - Added new routes to support Authoring page and Authoring/[resourceType]/[id] routes
 - Added mongodb connection file
 - Added mongodb basic read, create, and update functions
 - Added trpc router for read, create, and update for Library and Measure resources


# Testing guidance
 - add `MONGODB_URI=mongodb://localhost:27017/draft-repository` to `.env` file
 - `npm run check`
 - Fire up Robo3T
 - `npm run start:all`
 - Click on the Authoring button in the nav bar
 - Select either measure or library from the radio buttons and click `Create new Draft` button
 - Ensure that a proper notification appears, then check RoboMongo `draft-repository` collection to ensure that the resource was created
 - Play around with the url and identifier entry fields, and make sure that submitting the updates works as expected
 - Verify in robo3t that updates are actually going through
 - Try again with another resourceType
 - Take a close look at the setup for mongo and trpc and ensure that is all done as intended.
